src/charm.py

```diff
@@ -649,6 +676,70 @@ def _peers(self) -> Relation:
         """
         return self.model.get_relation(PEER)
 
+    def push_tls_files_to_workload(self, container: Container = None) -> None:
+        """Uploads TLS files to the workload container."""
+        if container is None:
+            container = self.unit.get_container("postgresql")
+
+        key, ca, cert = self.tls.get_tls_files()
+        if key is not None:
+            container.push(
+                f"{self._storage_path}/{TLS_KEY_FILE}",
+                key,
+                make_dirs=True,
+                permissions=0o400,
+                user=WORKLOAD_OS_USER,
+                group=WORKLOAD_OS_GROUP,
+            )
+        if ca is not None:
+            container.push(
+                f"{self._storage_path}/{TLS_CA_FILE}",
+                ca,
+                make_dirs=True,
+                permissions=0o400,
+                user=WORKLOAD_OS_USER,
+                group=WORKLOAD_OS_GROUP,
+            )
+        if cert is not None:
+            container.push(
+                f"{self._storage_path}/{TLS_CERT_FILE}",
+                cert,
+                make_dirs=True,
+                permissions=0o400,
+                user=WORKLOAD_OS_USER,
+                group=WORKLOAD_OS_GROUP,
+            )
+
+        self.update_config()
+
+    def _restart(self, _) -> None:
+        """Restart PostgreSQL."""
+        try:
+            self._patroni.restart_postgresql()
+        except RetryError as e:
+            logger.error("failed to restart PostgreSQL")
+            self.unit.status = BlockedStatus(f"failed to restart PostgreSQL with error {e}")
+
+    def update_config(self) -> None:
+        """Updates Patroni config file based on the existence of the TLS files."""
+        enable_tls = False
+        key, ca, cert = self.tls.get_tls_files()
+        if None not in [key, ca, cert]:
+            enable_tls = True
```

Comment:
nit, but I wonder if it's more explicit having a:

```python
enable_tls = all(self.tls.get_tls_files())
```

Reply:
Good catch! I will change it here. Thanks Marc!

Reply:
Updated on https://github.com/canonical/postgresql-k8s-operator/pull/36/commits/9e4d48110290a0002771b41ced72413d2d227dd1.